src/map/npc.cpp

```diff
@@ -5180,6 +5180,13 @@ void npc_parse_mob2(struct spawn_data* mob)
 	{
 		struct mob_data* md = mob_spawn_dataset(mob);
 		md->spawn = mob;
+		// Determine center cell for each mob in the spawn line
+		if (battle_config.randomize_center_cell) {
+			if (mob->xs > 1)
+				md->centerX = rnd_value(mob->x - mob->xs + 1, mob->x + mob->xs - 1);
```

Comment:
```suggestion
				md->centerX = rnd_value(mob->x - mob->xs + 1, mob->x + mob->xs - 1);
			else
				md->centerX = 0;
```

Reply:
Let's just leave centerX as initialized, then we also don't need the other changes!